build.rs

```diff
@@ -20,6 +20,152 @@ fn build_config_name() -> String {
     format!("mkl-dynamic-{}-{}", integer_config, parallelism)
 }
 
+/// Paths required for linking to MKL from MKLROOT folder
+struct MklDirectories {
+    #[allow(dead_code)]
+    mkl_root: String,
+    lib_dir: String,
+    omp_lib_dir: String,
+    include_dir: String,
+}
+
+impl MklDirectories {
+    /// Constructs paths required for linking MKL from the specified root folder. Checks if paths exist.
+    fn try_new(mkl_root: &str) -> Result<Self, String> {
+        let os = if cfg!(target_os = "windows") {
+            "win"
+        } else if cfg!(target_os = "linux") {
+            "lin"
+        } else {
+            return Err("Target OS not supported".into());
+        };
+
+        let arch = if cfg!(target_arch = "x86_64") {
+            "64"
+        } else {
+            return Err("Target architecture not supported".into());
+        };
+
+        let mkl_root: String = mkl_root.into();
+        let prefix: String = mkl_root.clone();
+        let exec_prefix: String = prefix.clone();
+        let lib_dir = format!(
+            "{exec_prefix}/lib/intel{arch}_{os}",
+            exec_prefix = exec_prefix,
+            arch = arch,
+            os = os
+        );
+        let omp_lib_dir = format!(
+            "{exec_prefix}/../compiler/lib/intel{arch}_{os}",
+            exec_prefix = exec_prefix,
+            arch = arch,
+            os = os
+        );
+        let include_dir = format!("{prefix}/include", prefix = prefix);
+
+        let mkl_root_path = PathBuf::from(mkl_root);
+        let lib_dir_path = PathBuf::from(lib_dir);
+        let omp_lib_dir_path = PathBuf::from(omp_lib_dir);
+        let include_dir_path = PathBuf::from(include_dir);
+
+        let mkl_root_str = mkl_root_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let lib_dir_str = lib_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let omp_lib_dir_str = omp_lib_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let include_dir_str = include_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+
+        // Check that paths exist
+
+        if !mkl_root_path.exists() {
+            return Err(format!(
```

Comment:
Is it possible to have "false positives" in this case? I.e., could it be that the path is valid in the sense of linking, but that it doesn't exist for some reason? I.e. something like a difference in working directories or something like that?

An alternative would be to simply warn if we detect that a path does not exist.

Reply:
I changed the errors to warnings.